### PR TITLE
[db] panic when writing db returns no space

### PR DESCRIPTION
**Pull request by huof6829**
# Description

User can define different partition path for chaindb/indexdb/triedb...   in config.yaml.  Monitor chaindb is not enough.
These dbs  are almost all large after a long time, so check writing db can be accurate.

Fixes https://github.com/iotexproject/iotex-core/issues/3629

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: db/db_bolt.go [hunk 1]
```diff
@@ -86,6 +88,9 @@ func (b *BoltDB) Put(namespace string, key, value []byte) (err error) {
 		}
 	}
 	if err != nil {
+		if errors.Is(err, syscall.ENOSPC) {
```
Comment (dustinxie) [inline db/db_bolt.go:91]: or should we use `errors.Cause(err) == syscall.ENOSPC`? can you look at bolt db source code to verify?

Comment (huof6829) [reply db/db_bolt.go:91]: `errors.Cause(err)` returns `write /run/data/chain.db: no space left on device`
`syscall.ENOSPC` is `no space left on device`
`errors.Is` means `syscall.ENOSPC` matches `err`

Comment (Liuhaai) [reply db/db_bolt.go:91]: errors.Is(erros.Cause(), err)

Comment (dustinxie) [reply db/db_bolt.go:91]:
```
// An error is considered to match a target if it is equal to that target or if
// it implements a method Is(error) bool such that Is(target) returns true.
func Is(err, target error) bool { return stderrors.Is(err, target) }
```
i think better to use `errors.Cause() == xxx` if possible? this is how error is decided in the codebase, try to be consistent

Comment (dustinxie) [reply db/db_bolt.go:91]:
```
// An error is considered to match a target if it is equal to that target or if
// it implements a method Is(error) bool such that Is(target) returns true.
func Is(err, target error) bool { return stderrors.Is(err, target) }
```

can you verify (or write a test case) this is the case?

File: server/itx/server.go [hunk 2]
```diff
@@ -232,20 +230,6 @@ func StartServer(ctx context.Context, svr *Server, probeSvr *probe.Server, cfg c
...
-	}
-	if err = m.Start(ctx); err != nil {
-		log.L().Panic("Failed to start monitor disk space.", zap.Error(err))
-	}
-	defer func() {
-		if err := m.Stop(ctx); err != nil {
-			log.L().Panic("Failed to stop monitor disk space.", zap.Error(err))
-		}
-	}()
-
```
Comment (dustinxie) [inline server/itx/server.go:248]: looks like you are reverting an earlier commit? can you open another PR to just revert that commit?

Comment (huof6829) [reply server/itx/server.go:248]: Ok. Split this pr into https://github.com/iotexproject/iotex-core/pull/3685 and https://github.com/iotexproject/iotex-core/pull/3686

**Pull request closed without merging**